src/Parallel/PupStlCpp17.hpp

```diff
@@ -0,0 +1,79 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <optional>
+#include <pup_stl.h>
+#include <utility>
+#include <variant>
+
+#include "Utilities/TMPL.hpp"
+
+namespace PUP {
+/// \ingroup ParallelGroup
+/// Serialization of std::optional for Charm++
+template <typename T>
+void pup(PUP::er& p, std::optional<T>& t) noexcept {  // NOLINT
+  bool valid = false;
+  if (p.isUnpacking()) {
+    p | valid;
+    if (valid) {
+      T value{};
+      p | value;
+      t = std::move(value);
+    } else {
+      t.reset();
+    }
+  } else {
+    valid = t.has_value();
+    p | valid;
+    if (valid) {
+      p | *t;
+    }
+  }
+}
+
+/// \ingroup ParallelGroup
+/// Serialization of std::optional for Charm++
+template <typename T>
+void operator|(er& p, std::optional<T>& t) noexcept {  // NOLINT
```

Comment:
Why is this just er& instead of PUP::er&?

Reply:
Because this function is added to the `PUP` namespace

Reply:
... but then why do you say pup::er& on line 76 in the analogous call frosted::variant? That looks to me like it's also in the PUP namespace

Reply:
Probably some historical thing about how I wrote the code. It should be removed. I will remove it :) Thanks!